server/services/return-of-service.js

```diff
@@ -99,8 +109,70 @@ const getReturnOfServiceStatuses = async ({ participantId }) =>
       }
     );
 
+const updateReturnOfServiceSite = async ({ participantId, newSiteId }) => {
+  await validateRosParticipant(participantId);
+  if (!newSiteId) {
+    throw new Error(rosError.fieldNotFound);
+  }
+  // Validate Site Id
+  const sites = await dbClient.db[collections.EMPLOYER_SITES].find({
+    'body.siteId': newSiteId,
+  });
+  if (sites.length === 0) {
+    throw new Error(rosError.noSiteAttached);
+  }
+
+  dbClient.db[collections.ROS_STATUS].update(
+    { participant_id: participantId, current: true },
+    {
+      site_id: newSiteId,
+    }
+  );
+};
+
+const updateReturnOfServiceDate = async ({ participantId, newDate }) => {
+  await validateRosParticipant(participantId);
+  if (!newDate) {
+    throw new Error(rosError.fieldNotFound);
+  }
+
+  dbClient.db[collections.ROS_STATUS].update(
+    { participant_id: participantId, current: true },
+    {
+      'data.date': newDate,
+    }
+  );
+};
+
+const updateReturnOfServiceStartDate = async ({ participantId, newStartDate }) => {
+  await validateRosParticipant(participantId);
+  if (!newStartDate) {
+    throw new Error(rosError.fieldNotFound);
+  }
+
+  // validate start date
+  const res = await dbClient.db[collections.ROS_STATUS].find({
```

Comment:
How is this validating the start date? It looks like we're validating a ROS status exists

Reply:
Maybe the wording is confusing: I'm looking for a participant who has `assigned-new-site` entry here. Basically someone who changed a site. If the entry is not present, then this participant never changed sites hence _Start date at a new site_ cannot be updated.

Reply:
It looks like this check is gone now, did it not end up being needed?